addons/mrp/models/mrp_production.py

```diff
@@ -480,7 +480,7 @@ def _compute_production_location(self):
             if production.product_id:
                 production.production_location_id = production.product_id.with_company(production.company_id).property_stock_production
             else:
-                production.production_location_id = location_by_company.get(production.company_id.id)[0]
+                production.production_location_id = location_by_company and location_by_company.get(production.company_id.id)[0]
```

Comment:
I would rather replace the for-loop body with
```py
prod_loc = production.product_id.with_company(production.company_id).property_stock_production
comp_locs = location_by_company.get(production.company_id.id)
production.production_location_id = prod_loc or (comp_locs and comp_locs[0])
```
Because currently the `else` content seems useless: as soon as the production has a product (which is a required field), we will never enter in that `else` anymore. It means that, when we are in it (when the product is still empty), we are in a temporary stage (the product will be soon defined and the compute will be executed again) and the current value does not matter
Worst: if the product does not have any `property_stock_production`, we will not use the company location as fallback.